tests/specs/lemmas.k

```diff
@@ -241,6 +241,9 @@ module LEMMAS-HASKELL [symbolic, kore]
     imports EVM
     imports EDSL
 
+    rule #lookup ( M:Map [ K1 <- V1 ] , K2 ) => #lookup ( K1 |-> V1 , K1 ) requires K1 ==Int  K2 [simplification]
+    rule #lookup ( M:Map [ K1 <- _  ] , K2 ) => #lookup ( M         , K2 ) requires K1 =/=Int K2 [simplification]
+
```

Comment:
Note that the first rule relies on the original #lookup rules in the semantics, to handle the different cases, in the same way, no matter how they will be changed in the future.